### ISSUE TITLE
Process alignments in chunks

### ISSUE THREAD

**Issue by evgenyrp** [mozilla/translations#739]
It will be cheaper to split it into chunks and run on smaller preemptible machines rather than one big standard instance. The downside is that it will increase the complexity of the graph and will be harder to maintain.

**Comment by eu9ene** [comment]
Another approach would be to process it in chunks on one machine and if it's preempted, continue from the last unprocessed chunk. This approach can work but it takes longer to process compared to parallelization. One more thing to take into account is that we need to calculate priors on a large part of the original parallel corpus first before we can start processing anything in smaller chunks. I think a 100M sample might be sufficient.

**Comment by eu9ene** [comment]
Chunking on one machine is already implemented #715

**Comment by gregtatum** [comment]
We may be able to implement chunking in a generalized fashion using a transform. We could automate the chunking and dechunking with declarative yaml. The graph would be more complicated, but the kind files would become simpler.

**Comment by ZJaume** [comment]
Maybe it is not complicated to modify eflomal to work with compressed files? [Here's](https://github.com/robertostling/eflomal/blob/1fe2a43e3667fb2461736ddb48783ab0cf98a171/python/scripts/eflomal-align#L138-L149) where it opens the input files and [here](https://github.com/robertostling/eflomal/blob/1fe2a43e3667fb2461736ddb48783ab0cf98a171/python/eflomal/__init__.py#L65) the output files. We could just write a function that opens a normal file or a compressed file depending on the filename extension or a cli option. Zstd library has an [open](https://python-zstandard.readthedocs.io/en/latest/misc_apis.html#open) function that returns a file object.

**Comment by gregtatum** [comment]
https://github.com/robertostling/eflomal/blob/1fe2a43e3667fb2461736ddb48783ab0cf98a171/python/eflomal/__init__.py#L65-L67

It also writes them directly back out again to disk.

**Comment by evgenyrp** [comment]
Here's a distillation corpus alignment task for 676M German dataset https://firefox-ci-tc.services.mozilla.com/tasks/fcJ9qBg-Th-bmk_4w9m1Vw. It takes a while on a non-preemptible machine (4+ days).

**Comment by evgenyrp** [comment]
I optimized it to work more efficiently on one machine. I'll test it while doing the big retraining; maybe we don't need to complicate the pipeline by splitting it into multiple tasks. If it's still too slow and we need to do that, the splitting code is already there, so it's just Taskcluster-related work.
